Lib/inspect.py

```diff
@@ -2241,9 +2241,13 @@ def _signature_from_callable(obj, *,
                 sigcls=sigcls)
 
             sig = _signature_get_partial(wrapped_sig, partialmethod, (None,))
-
             first_wrapped_param = tuple(wrapped_sig.parameters.values())[0]
-            new_params = (first_wrapped_param,) + tuple(sig.parameters.values())
+            sig_params = tuple(sig.parameters.values())
+            if first_wrapped_param.kind is Parameter.VAR_POSITIONAL:
+                new_params = sig_params
```

Comment:
Just realized that it's a bit inefficient to construct a new signature objects with a copy of its parameters. Put a `return sig` here...